scality_glance_store/store.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) 2015 Scality
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import hashlib
+import logging
+
+from glance_store import backend
+from glance_store import capabilities
+from glance_store.common import utils
+from glance_store import driver
+from glance_store import exceptions
+from glance_store.i18n import _, _LI, _LE
+from glance_store import location
+
+from oslo_config import cfg
+from oslo_utils import excutils
+from oslo_utils import units
+
+import scality_sproxyd_client.exceptions
+from scality_sproxyd_client import sproxyd_client
+import scality_sproxyd_client.utils
+
+
+LOG = logging.getLogger(__name__)
+logging.getLogger('urllib3.util.retry').level = logging.INFO
+
+_SPROXYD_OPTS = [
+    cfg.ListOpt('scality_sproxyd_endpoints',
+                help=_("Comma-separated list of Sproxyd endpoints which "
+                       "accept queries 'by path' (e.g. 'http://10.5.9.2:81/"
+                       "proxy/chord_path/')")),
+]
+
+
+class StoreLocation(location.StoreLocation):
+    """
+    Class describing a relative Sproxyd URI.
+
+    In the form of: scality://image
+    """
+
+    def process_specs(self):
+        self.image_id = self.specs['image_id']
+
+    def get_uri(self):
+        return "scality://%s" % self.image_id
+
+    def parse_uri(self, uri):
+        prefix = 'scality://'
+        pieces = uri[len(prefix):].split('/')
+        if not uri.startswith(prefix) or len(pieces) > 1 or not pieces[0]:
+            msg = _('Invalid URI : %s. URI must be of the form '
+                    'scality://image') % uri
+            LOG.info(msg)
+            raise exceptions.BadStoreUri(message=msg)
+
+        self.image_id = pieces[0]
+
+
+class Store(driver.Store):
+
+    _CAPABILITIES = (capabilities.BitMasks.RW_ACCESS |
+                     capabilities.BitMasks.DRIVER_REUSABLE)
+    CHUNKSIZE = 64 * units.Ki
+    OPTIONS = _SPROXYD_OPTS
+
+    def __init__(self, conf):
+        super(Store, self).__init__(conf)
+
+        endpoints = self.conf.glance_store.scality_sproxyd_endpoints
+        self.sproxyd_client = sproxyd_client.SproxydClient(endpoints)
+
+    @staticmethod
+    def get_schemes():
+        return ('scality',)
```

Comment:
Is this duplication with `StoreLocation`'s `scheme`? If so, consider having this string somewhere defined globally in this module.